src/codegen/doc/format.rs

```diff
@@ -77,32 +77,51 @@ fn format(mut input: &str, symbols: &symbols::Info) -> String {
 }
 
 static SYMBOL: Lazy<Regex> =
-    Lazy::new(|| Regex::new(r"(^|[^\\])[@#%]([\w]+\b)([:.]+[\w_-]+\b)?").unwrap());
+    Lazy::new(|| Regex::new(r"(^|[^\\])([@#%])(\w+\b)([:.]+[\w-]+\b)?").unwrap());
 static FUNCTION: Lazy<Regex> = Lazy::new(|| Regex::new(r"(\b[a-z0-9_]+)\(\)").unwrap());
-static GDK_GTK: Lazy<Regex> = Lazy::new(|| Regex::new(r"G[dt]k[A-Z][\w]+\b").unwrap());
+static GDK_GTK: Lazy<Regex> = Lazy::new(|| Regex::new(r"G[dt]k[A-Z]\w+\b").unwrap());
 static TAGS: Lazy<Regex> = Lazy::new(|| Regex::new(r"<[\w/-]+>").unwrap());
-static SPACES: Lazy<Regex> = Lazy::new(|| Regex::new(r"[ ][ ]+").unwrap());
+static SPACES: Lazy<Regex> = Lazy::new(|| Regex::new(r"[ ]{2,}").unwrap());
 
 fn replace_c_types(entry: &str, symbols: &symbols::Info) -> String {
-    let lookup = |s: &str| -> String {
-        symbols
-            .by_c_name(s)
-            .map(symbols::Symbol::full_rust_name)
-            .unwrap_or_else(|| s.into())
-    };
+    let lookup_opt =
+        |s: &str| -> Option<String> { symbols.by_c_name(s).map(symbols::Symbol::full_rust_name) };
+    let lookup = |s: &str| -> String { lookup_opt(s).unwrap_or_else(|| s.into()) };
+
     let out = SYMBOL.replace_all(entry, |caps: &Captures<'_>| {
-        format!(
-            "{}`{}{}`",
-            &caps[1],
-            lookup(&caps[2]),
-            caps.get(3).map(|m| m.as_str()).unwrap_or("")
-        )
+        let trail = caps.get(4).map(|m| m.as_str()).unwrap_or("");
+        match &caps[2] {
+            // Parameter (and member?) names
+            // TODO: If member names, these can probably be emitted for simple structs where the fields are pub?
+            "@" => format!("{}`{}{}`", &caps[1], &caps[3], trail),
+            // Global symbols
+            "#" => {
+                if let Some(sym) = lookup_opt(&caps[3]) {
+                    format!("{}[`{n}{t}`](crate::{n}{t})", &caps[1], n = sym, t = trail)
+                } else {
+                    format!("{}`{}{}`", &caps[1], &caps[3], trail)
+                }
+            }
+            // Global constants
+            "%" => {
+                if let Some(sym) = lookup_opt(&caps[3]) {
+                    format!("{}[`{}{}`]", &caps[1], sym, trail)
+                } else {
+                    format!("{}`{}{}`", &caps[1], &caps[3], trail)
+                }
+            }
+            u => panic!("Unknown reference type {}", u),
+        }
     });
     let out = GDK_GTK.replace_all(&out, |caps: &Captures<'_>| {
         format!("`{}`", lookup(&caps[0]))
     });
     let out = FUNCTION.replace_all(&out, |caps: &Captures<'_>| {
-        format!("`{}`", lookup(&caps[1]))
+        if let Some(func) = lookup_opt(&caps[1]) {
+            format!("[`{f}()`](crate::{f}())", f = &func)
+        } else {
+            format!("`{}()`", &caps[1])
```

Comment:
Added `()` here:

- It is already in the docs;
- More clear when reading the docs, knowing that this references a function and not some other symbol;
- Clearer error messages by `rustdoc`;
- Rustdoc probably also enforces the link to point to a function, and fails otherwise (not tested).